pkg/contracts/src/CVStrategy/facets/CVStreamingFundingFacet.sol

```diff
@@ -0,0 +1,304 @@
+        );
+    }
+
+    /**
+     * @notice Batch evaluate all active funding proposals
+     * @dev Keeper function - efficiently evaluates all proposals
+     */
+    function batchEvaluateStreams() external {
+        _checkOnlyInitialized();
+        
+        if (proposalType != ProposalType.Funding && proposalType != ProposalType.Streaming) {
+            revert WrongProposalType();
+        }
+        
+        for (uint256 i = 1; i <= proposalCounter; i++) {
+            if (proposals[i].proposalStatus == ProposalStatus.Active) {
+                this.evaluateProposalStream(i);
+            }
+        }
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                        INTERNAL HELPERS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Calculate threshold for a funding proposal
+     * @param proposalId Proposal ID
+     * @return uint256 Required conviction threshold
+     */
+    function _calculateProposalThreshold(uint256 proposalId) internal view returns (uint256) {
+        Proposal storage p = proposals[proposalId];
+        
+        if (p.requestedAmount == 0) {
+            return 0; // No threshold for zero amount
+        }
+        
+        return ConvictionsUtils.calculateThreshold(
+            p.requestedAmount,
+            getPoolAmount(),
+            totalPointsActivated,
+            cvParams.decay,
+            cvParams.weight,
+            cvParams.maxRatio,
+            cvParams.minThresholdPoints
+        );
+    }
+
+    /**
+     * @notice Start funding stream with calculated units
+     * @param proposalId Proposal to start streaming to
+     */
+    function _startFundingStream(uint256 proposalId) internal {
+        Proposal storage p = proposals[proposalId];
+        
+        // Calculate units for funding proposal
+        // Strategy: Allocate fixed units or proportional to requested amount
+        uint128 units = _calculateFundingUnits(proposalId);
+        
+        if (units > 0) {
+            _startStreamViaCore(proposalId, p.beneficiary, units);
+        }
+    }
+
+    /**
+     * @notice Calculate units for a funding proposal
+     * @dev Multiple strategies possible:
+     *      1. Fixed units per proposal (simple)
+     *      2. Proportional to requestedAmount vs pool size
+     *      3. Based on conviction level above threshold
+     * @param proposalId Proposal ID
+     * @return uint128 Units to allocate
+     */
+    function _calculateFundingUnits(uint256 proposalId) internal view returns (uint128) {
+        Proposal storage p = proposals[proposalId];
+        uint256 poolSize = getPoolAmount();
+        
+        if (poolSize == 0) return 0;
+        
+        // Strategy: Proportional to requested amount
+        // e.g., requesting 10% of pool → get 10% of flow (1000 units out of 10000)
+        uint256 requestRatio = (p.requestedAmount * 10000) / poolSize;
```

Comment:
we should definitly not have the requested amount field in a streaming pool, I would go with the 3rd strategy